File: android/src/main/java/xyz/justzappit/zappmessaging/models/ZMIdentity.kt
package xyz.justzappit.zappmessaging.models

import kotlinx.serialization.Serializable

/**
 * User's cryptographic identity.
 * Mirrors iOS ZMIdentity for cross-platform consistency.
 */
@Serializable
data class ZMIdentity(
    /** Ed25519 public key (hex encoded) */
    val publicKey: String,
    /** User's display name */
    val displayName: String,
    /** When the identity was created (epoch millis) */
    val createdAt: Long = System.currentTimeMillis()
)

File: ios/Sources/ZappMessaging/Models/ZMIdentity.swift
//
//  ZMIdentity.swift
//  ZappMessaging
//
//  Represents a user's cryptographic identity
//

import Foundation

/// User's cryptographic identity
public struct ZMIdentity: Codable, Identifiable, Equatable, Sendable {
    /// Unique identifier (same as publicKey)
    public var id: String { publicKey }
    
    /// Ed25519 public key (hex encoded)
    public let publicKey: String
    
    /// User's display name
    public let displayName: String
    
    /// When the identity was created
    public let createdAt: Date
    
    public init(publicKey: String, displayName: String, createdAt: Date = Date()) {
        self.publicKey = publicKey
        self.displayName = displayName
        self.createdAt = createdAt
    }
}
